contrib/cmd/fs-idmap/fs-idmap.go

```diff
@@ -11,18 +11,18 @@ import (
 )
 
 func main() {
-	if len(os.Args) < 2 {
+	if len(os.Args) != 2 {
 		log.Fatalf("usage: %s path_to_mount_set_attr", os.Args[0])
 	}
 
 	src := os.Args[1]
-	treeFD, err := unix.OpenTree(-1, src, uint(unix.OPEN_TREE_CLONE|unix.OPEN_TREE_CLOEXEC|unix.AT_EMPTY_PATH|unix.AT_RECURSIVE))
+	treeFD, err := unix.OpenTree(-1, src, uint(unix.OPEN_TREE_CLONE|unix.OPEN_TREE_CLOEXEC|unix.AT_EMPTY_PATH))
 	if err != nil {
 		log.Fatalf("error calling open_tree %q: %v", src, err)
 	}
 	defer unix.Close(treeFD)
 
-	cmd := exec.Command("/usr/bin/sleep", "5")
+	cmd := exec.Command("sleep", "5")
```

Comment:
I think a better way of doing this is to use `Ptrace: true` to avoid doing an exec at all (that's what I'm doing in #3953). But I guess since this is a contrib program it doesn't really matter.

Reply:
I agree it is not worth it here.

But how is it that having ptrace true in the `SysProcAttr` avoids an exec?

Reply:
`Ptrace: true` causes the Go stdlib to do `PTRACE_TRACEME` [which blocks the spawned process immediately before the exec](https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/syscall/exec_linux.go;l=605) (waiting for a tracer to attach to it).

Reply:
I might use this for containerd, great tip! :)